### PR TITLE
[Math] Optimize Matrix mult and invert

### PR DESCRIPTION
**Pull request by Eideren**
# PR Details
Passing operations to numerics where appropriate to benefit from SIMD, only did it for the two most commonly used function right now, we could look into writing it explicitly for column major later.
Also small cleanup for `TransformProcessor`.

## Types of changes
- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist
- [ ] My change requires a change to the documentation.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: sources/core/Stride.Core.Mathematics/Matrix.cs [hunk 1]
```diff
@@ -1014,39 +1017,42 @@ public static void Multiply(ref Matrix left, ref Matrix right, out Matrix result
...
-        /// <returns>The product of the two matrices.</returns>
-        public static Matrix Multiply(Matrix left, Matrix right)
+        /// <param name="result">The product of the two matrices.</param>
+        [Obsolete($"Use {nameof(Multiply)} instead")]
+        public static void MultiplyRef(ref Matrix left, ref Matrix right, ref Matrix result)
         {
-            Matrix result;
             Multiply(ref left, ref right, out result);
-            return result;
         }
```
Comment (Eideren) [inline sources/core/Stride.Core.Mathematics/Matrix.cs:1046]: Small diff fail here, `Matrix Multiply(Matrix left, Matrix right)` hasn't been removed and `MultiplyRef` isn't new, the former is now right below and the later was above but has been replaced.

**Comment by tebjan** [comment]
this looks very interesting, my only concern is that the conversion overhead is making the improvement negligible. can you see in a benchmark that this is actually faster?

**Comment by Eideren** [comment]
With a cpu supporting the SSE set multiply is ~40% faster and invert is ~33% faster.
Overhead has been mostly optimized out, stripping that bloat out leads to ~7% additional improvement but I'm not sure that's worth it considering dot net might improve support while we aren't looking.
Here's a [sharplab](https://sharplab.io/#v2:C4LghgzgtgPgAgBgAQFkzAE4EsAeARAe2ADsBTYJAXiTgEYA6AOQFcpTsBjCetTXAFhz8A3AFgAUIhoMASs2LAsbegGECUAA5YANuwDK7AG5YOpCGPESImZhwq9sOCQG8JSdzQDMSAGbaC6Ki0tBYeXr7+gSgATCFuHnDefgH2nnHiYYkRKaj86ZlJkfa00aEJhTkxpfHuWclRntUZ5dlR/E0FrcWeZbUVUdE9NeH1qUPNfV25452jQSLDdUWo0QsTI8sonmuzm/w7HsOL3hwExNZIAEYEBNpIADJgAJ4EzMAAkhAyBADuaABWBAwVF8YG0EFIvSQi1oADYkBhSD5UOhHIQSOQkABVc5gHykGRIgCCEHRjHIAAosMQUXwcEgoABKKgAPgRSOxuPx9BJAB4HLgADS0tFEMjAFkUxHInEQPGkHlfJFUmlMxlQujw6Ui3CcuX4wk+ABiGHUZMp1J1+DFmKZrPZMq5Cr5AutGOAwtdku1svlisNKoZjPVR3Wx2k8Lg/FQzG0ig02ieAHltAATKUc11IXQ+D0Oq0IrAAcwAFnnXvZUbrERBY8BGcNXOswjW6zxgiCKTngO3aEgAFSF0s9lDBZkAaiQXaRI5KA6HZZ4sQnU+77e8g+ww54aRX09z7ejm+Li5QeRDzY8rbjS771H3I9i863p7HSEnD6X0WfJ8ftD3a4xBuC4jru76rjOS5HiBPDnlCLZmG2Wx3hBB7IT+26jv+4Gfls37Hphy44YBnjAS+oHYR+JHQeRsH/vBV6ITeZ4obheQYa+lGoSO7QcX+AGQWeZG/juXFsTRIksReYQIbWzFzvegEdgRr7RAJaFzipj5qcRgm0MJmGNOps4SZh7TSTJ7jXtpnaAU+WntjpVGCdE+EwVUxlLgZp5GbpaGrHxsFqQxVlMaB36KYJaSBaOTncTubm0R5fmgd54WeVspmnuZIXsnJPERfFLExSUGW8Q5yXOWh/BpQlZVZQVFmWdZ662XpfYVfpGWae5gzdbVWyeN1DWwUNuUtUBbX+R17ldSlX4xX182TRVpEZQFFU1U1MkTaRU0USVQ3LY0i1HVVqUxWtx0jUJ22yUhNX7XRh1lYlklLedo2XWdRX7DFW3jWFh5PVhJX8N1b2EeDy36Zd0OfaON37HdjH5VBIP2bN8NFa5i3Y7hgxw+tSPg4DaOZSD0WdfjJGQ6eqwZXtq001FJMo6F5N/ZF1UzUltAs9VdOPgLPEDds9X/aTwwAL6hp0mo0NGKB1lgCZPO8xCBlm3bCpaWbkcKFYFtZDbrE2lmhcirrouK2Ygj6TqGiS5rAIG3bs/m1s2hQwLUA7+oEsSpJEOSruWuRHu+tyegANaqxrWCu0bJu5VHgc+M7IcWjSJsgo8LxvJ83x/GAgLAgA/Hbm5ICACLztoUKy2G6xZArWYEBo7DoECA5a1W9I60gev9yBpthObFval77p237HJp07wfAKHbszh7U/9zbmK+/mC9By7gYR7lcAAOx6vKhommaWeu/nFZF78AI95XdzV7XwKDto21N2E4YK1GSANaGHYK7DedIkCGDBMwUghs3jGzCmPDwE9LJp3oLHeOxBE4UmTgg3K09xT0CASAike8M5LxXpA7Q0DGSwIoKQzOy9KQm2/hIaWQA=) setup if you want to take a look at it.

**Comment by ykafia** [comment]
Would it be safer to use [this library](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.intrinsics?view=net-6.0) ? 
It seems to work only on ARM et x86 architectures though

**Comment by Eideren** [comment]
That's what numerics uses internally, so technically we already are using that. What do you mean by safer ? If you mean using them directly instead of passing through dotnet's types, sure, but as I said above I'm not sure that's worth it considering dot net might improve support while we aren't looking.

**Comment by tebjan** [comment]
> multiply is ~40% faster and invert is ~33% faster.
> Overhead has been mostly optimized out, stripping that bloat out leads to ~7% additional improvement

that sounds great! i am ok with merging this, if it will cause any problems, we can still revert it.

**Pull request merged**